lib/client/client.go

```diff
@@ -166,6 +166,104 @@ func (proxy *ProxyClient) GenerateCertsForCluster(ctx context.Context, routeToCl
 	return trace.Wrap(err)
 }
 
+// ReissueParams encodes optional paramters for
+// user certificate reissue.
+type ReissueParams struct {
+	RouteToCluster string
+	AccessRequests   []string
+}
+
+// ReissueUserCerts generates certificates for the user
+// that have a metadata instructing server to route the requests to the cluster
+func (proxy *ProxyClient) ReissueUserCerts(ctx context.Context, params ReissueParams) error {
+	localAgent := proxy.teleportClient.LocalAgent()
+	key, err := localAgent.GetKey()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	cert, err := key.SSHCert()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	tlsCert, err := key.TLSCertificate()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	clusterName, err := tlsca.ClusterName(tlsCert.Issuer)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	clt, err := proxy.ConnectToCluster(ctx, clusterName, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if params.RouteToCluster != "" {
+		// Before requesting a certificate, check if the requested cluster is valid.
+		_, err = clt.GetCertAuthority(services.CertAuthID{
+			Type:       services.HostCA,
+			DomainName: params.RouteToCluster,
+		}, false)
+		if err != nil {
+			return trace.NotFound("cluster %v not found", params.RouteToCluster)
+		}
+	}
+	req := proto.UserCertsRequest{
+		Username:       cert.KeyId,
+		PublicKey:      key.Pub,
+		Expires:        time.Unix(int64(cert.ValidBefore), 0),
+		RouteToCluster: params.RouteToCluster,
+		AccessRequests:   params.AccessRequests,
```

Comment:
Strange, did `gofmt` miss this? The rest of the file looks good.

Reply:
Ah, I have `gofmt` auto-applied when I exit vim; probably a result of force-quitting for some reason.